src/components/MediaRecorder/classes/MediaRecorderController.ts

```diff
@@ -0,0 +1,353 @@
+import fixWebmDuration from 'fix-webm-duration';
+import { nanoid } from 'nanoid';
+import {
+  createFileFromBlobs,
+  getExtensionFromMimeType,
+  getRecordedMediaTypeFromMimeType,
+  RecordedMediaType,
+} from '../../ReactFileUtilities';
+import { mergeDeep } from '../../../utils/mergeDeep';
+import { transcode } from '../transcode';
+import { resampleWaveformData } from '../../Attachment';
+import { isSafari } from '../../../utils/browsers';
+import { BrowserPermission } from './BrowserPermission';
+import { TranslationContextValue } from '../../../context';
+import { defaultTranslatorFunction } from '../../../i18n';
+import { Subject } from '../observable/Subject';
+import { BehaviorSubject } from '../observable/BehaviorSubject';
+import {
+  AmplitudeAnalyserConfig,
+  AmplitudeRecorder,
+  AmplitudeRecorderConfig,
+} from './AmplitudeRecorder';
+
+import type { LocalVoiceRecordingAttachment } from '../../MessageInput';
+
+const RECORDED_MIME_TYPE_BY_BROWSER = {
+  audio: {
+    others: 'audio/webm',
+    safari: 'audio/mp4;codecs=mp4a.40.2',
```

Comment:
Do we convert mp4/aac to mp3 as well?

Reply:
true, that is unnecessary

Reply:
https://github.com/GetStream/stream-chat-react/pull/2339/commits/f78ef717ce8d34113572dd3aa8ba373fdb7be682